src/app/shared/couchdb.service.ts

```diff
@@ -82,4 +82,35 @@ export class CouchService {
     return obs;
   }
 
+  newPrepAttachment(file) {
+    file = file.replace(/^data:image\/\w+;base64,/gi, '');
+    const byteCharacters = atob(file);

```

Comment:
This is not obvious to a reader of the code, and should probably not be without commenting. If it is unclear exactly what each step does to you, at least give a reference to where you found the code snippets https://stackoverflow.com/questions/16245767/creating-a-blob-from-a-base64-string-in-javascript